spectacles/validators.py

```diff
@@ -73,21 +74,55 @@ def __init__(self, client: LookerClient, project: str):
         super().__init__(client)
         self.project = project
 
-    def validate(self) -> Dict[str, Any]:
-        tests = self.client.all_lookml_tests(self.project)
+    def validate(
+        self,
+        selectors: Optional[List[str]] = None,
+        exclusions: Optional[List[str]] = None,
+    ) -> Dict[str, Any]:
+        # Assign default values for selectors and exclusions
+        if selectors is None:
+            selectors = ["*/*"]
+        if exclusions is None:
+            exclusions = []
 
-        # The error objects don't contain the name of the explore
-        # We create this mapping to help look up the explore from the test name (unique)
-        test_to_explore = {test["name"]: test["explore_name"] for test in tests}
+        all_tests = self.client.all_lookml_tests(self.project)
+        selected_tests = []
+        test_to_explore = {}
+        for test in all_tests:
+            if is_selected(
+                test["model_name"], test["explore_name"], selectors, exclusions
+            ):
+                selected_tests.append(test)
+                # The error objects don't contain the name of the explore
+                # We create this mapping to help look up the explore from the test name
+                test_to_explore[test["name"]] = test["explore_name"]
+
+        test_count = len(selected_tests)
+        if test_count == 0:
+            raise SpectaclesException(
+                name="no-data-tests-found",
+                title="No data tests found.",
+                detail=(
+                    "If you're using --explores or --exclude, make sure your project "
+                    "has data tests that reference those models or explores."
+                ),
+            )
 
-        test_count = len(tests)
         printer.print_header(
             f"Running {test_count} {'test' if test_count == 1 else 'tests'}"
         )
 
+        test_results: List[Dict[str, Any]] = []
+        for test in selected_tests:
+            test_name = test["name"]
+            model_name = test["model_name"]
+            results = self.client.run_lookml_test(
+                self.project, model=model_name, test=test_name
+            )
+            test_results.extend(results)
+
```

Comment:
I don't think there's anything we can do about this, but this will kick them off sequentially, right?

In theory, if a whole model is going to be tested, we could test it in one go I think, which would get us a decrease in run time because Looker would kick them off simultaneously. I think to do this we'd have to change the selection logic though.

Reply:
Yeah, the most we could do with the endpoint would be to run them on a model level and then filter the results. Didn't seem worth the lift to me until there's demonstrated performance issues.